src/test/extension.test.ts

```diff
@@ -0,0 +1,56 @@
+import expect from 'expect'
+import mock from 'mock-require'
+import { createMockSourcegraphAPI } from './stubs'
+
+const sourcegraph = createMockSourcegraphAPI()
+// For modules importing Range/Location/Position/URI/etc
+mock('sourcegraph', sourcegraph)
+
+import { decorateLine } from '../extension'
+
+describe('extension', () => {
+    it('works', () => void 0)
+})
+
+const index = 5
+const pattern = /throw new Error+\(['"]([^'"]+)['"]\)/gi
+const line = "            throw new Error('cannot determine file path')"
+const match = pattern.exec(line)
+const noMissingConfigData: string[] = []
+const missingConfigData = ['repoMatch']
+const sentryProjectId = '134412'
+
+const decorationSuccess = {
+    range: new sourcegraph.Range(5, 0, 5, 0),
+    isWholeLine: true,
+    after: {
+        backgroundColor: '#e03e2f',
+        color: 'rgba(255, 255, 255, 0.8)',
+        contentText: ' View logs in Sentry » ',
+        hoverMessage: ' View logs in Sentry » ',
+        linkURL:
+            'https://sentry.io/organizations/sourcegraph/issues/?project=134412&query=is%3Aunresolved+cannot%20determine%20file%20path&statsPeriod=14d',
+    },
+}
+
+const decorationIncomplete = {
+    range: new sourcegraph.Range(5, 0, 5, 0),
+    isWholeLine: true,
+    after: {
+        backgroundColor: '#f2736d',
+        color: 'rgba(255, 255, 255, 0.8)',
+        contentText: ' View logs in Sentry (❕)» ',
+        hoverMessage:
+            ' Please fill out the following configurations in your Sentry extension settings: ' + missingConfigData,
+        linkURL:
+            'https://sentry.io/organizations/sourcegraph/issues/?project=134412&query=is%3Aunresolved+cannot%20determine%20file%20path&statsPeriod=14d',
+    },
+}
+
+describe('decorate line', () => {
+    it('decorates the line when no configs are missing', () =>
+        expect(decorateLine(index, match!, noMissingConfigData, sentryProjectId)).toEqual(decorationSuccess))
```

Comment:
Could you just pass an array of strings to `decorateline()` rather than having to declare `pattern`, `line` and perform a regex match just to set up your test case? This would allow you to avoid non-null assertion usage on `match`, and also to test more cases with less setup code.